### ISSUE TITLE
Add a number of reruns for failed tests

### ISSUE THREAD

**Issue by zakaprov** [golemfactory/goth#545]
`yagna` integration tests are inherently non-deterministic as they may fail due to external factors like network connectivity (timeouts, incomplete downloads etc.).
A potential remedy for this would be to introduce a limited number of retries on failed tests, for example via `pytest` by using the plugin [`pytest-rerunfailures`](https://pypi.org/project/pytest-rerunfailures/).

**Comment by johny-b** [comment]
Looks like `pytest-rerunfailures` will not work because it doesn't work with `pytest-asyncio`: https://github.com/pytest-dev/pytest-rerunfailures/issues/154

**Comment by johny-b** [comment]
OK, I did some research.
1. I tried `pytest-rerunfailures` and `flaky` (other pluging doing the same) and neither of them works with `pytest.asyncio`. I don't know about any other such plugin.
2. I tried some stupid fixes like changing the scope of `event_loop` fixture to `session` and it didn't work
3. I think `pytest-rerunfixtures` doesn't work with `session` fixtures either (together with package/module/class fixtures mentioned in the docs). At least I tried:
```

@pytest.fixture(scope="session", autouse=True)
def ttt():
    print("TTTT")
```
and there was more than one TTTT in the logs. Flaky does the same.
4. I check the implementation of `flaky`/`pytest-rerunfailures` (https://github.com/pytest-dev/pytest-rerunfailures/blob/master/pytest_rerunfailures.py) and I think that we probably **could** do something similar inside `goth` - this doesn't look very hard. Although there's a risk that doing this & maintaining compatibility with all other plugins might be hard.
5. I can think about some stupid-but-working fixes. E.g. we could repeat each test 5 times, but skip it if previous run was successful. I'm against this because it's ugly, but I could consider this if we can't find any other solution.
6. The best solution would be to fix `pytest-rerunfailures` : )
7. I'll try one more not-pretty-but-quite-harmless patch more : )
